### PR TITLE
Api profile

### PR DESCRIPTION
**Pull request by ninachurch**
functional call to api page, returns data into template

### CODE REVIEW

File: source/public/javascripts/apiprofilepage.js [hunk 1]
```diff
@@ -5,43 +5,53 @@ app.ApiProfile = {
...
-    this.$el.html(this.template(this.model.attributes));
+    console.log(this.id)
+    var apiObject = new this.model
+    var that = this;
+    $.ajax({
+      url:'/apis/' + this.id,
+      success: function(result){
+        apiObject.set({title: result.api.title, average_score: result.api.average_score, data_type: result.api.data_type, description: result.api.description, doc_link: result.api.doc_link, endpoint_link: result.api.endpoint_link, logo_url: result.api.logo_url, num_followers: result.api.num_followers, tips: result.api.tips});
+        that.$el.html(that.template(apiObject.attributes));
+            debugger
```
Comment (yohanyi) [inline source/public/javascripts/apiprofilepage.js]: REMOVEE!

File: source/public/javascripts/apiprofilepage.js [hunk 2]
```diff
@@ -5,43 +5,53 @@ app.ApiProfile = {
...
   render: function() {
-    console.log(this.model.attributes.logo_url)
-    this.$el.html(this.template(this.model.attributes));
+    console.log(this.id)
+    var apiObject = new this.model
+    var that = this;
+    $.ajax({
+      url:'/apis/' + this.id,
+      success: function(result){
+        apiObject.set({title: result.api.title, average_score: result.api.average_score, data_type: result.api.data_type, description: result.api.description, doc_link: result.api.doc_link, endpoint_link: result.api.endpoint_link, logo_url: result.api.logo_url, num_followers: result.api.num_followers, tips: result.api.tips});
```
Comment (Lkroner) [inline source/public/javascripts/apiprofilepage.js]: Change to result.api

File: source/public/javascripts/apiprofilepage.js (hunk 1 above)
Comment (Lkroner) [inline source/public/javascripts/apiprofilepage.js]: Take out debugger

File: source/public/javascripts/apiprofilepage.js [hunk 3]
```diff
@@ -5,43 +5,53 @@ app.ApiProfile = {
...
-  model: new app.ApiProfile.Models.Profile,
+  model: app.ApiProfile.Models.Profile,
 
-  events: {
-    // event listeners for info tab, tips/tricks and links
-  },
-
-  initialize: function(){
+  initialize: function(opts){
     console.log("Api Profile view initialized")
```
Comment (Lkroner) [inline source/public/javascripts/apiprofilepage.js]: Take out console log

File: source/public/javascripts/apiprofilepage.js [hunk 4]
```diff
@@ -5,43 +5,53 @@ app.ApiProfile = {
...
+    this.id = opts.id
   },
 
   template: _.template($('#apiprofile-template').html()),
 
   render: function() {
-    console.log(this.model.attributes.logo_url)
-    this.$el.html(this.template(this.model.attributes));
+    console.log(this.id)
```
Comment (Lkroner) [inline source/public/javascripts/apiprofilepage.js]: Take out console.log

File: source/public/javascripts/searchresults.js [hunk 5]
```diff
@@ -17,14 +20,23 @@ app.SearchResults.View = Backbone.View.extend({
...
     }.bind(this)).fail(function(){
     })
   },
 
   render: function(){
     this.ajaxRequest();
+  },
+
+  navigateToProfile: function(){
+    console.log("clicked search result")
```
Comment (Lkroner) [inline source/public/javascripts/searchresults.js]: Take out console log

File: source/public/javascripts/apiprofilepage.js [hunk 6]
```diff
@@ -5,43 +5,53 @@ app.ApiProfile = {
...
+    "description": "",
+    "tips": "",
+    "average_score": "",
+    "logo_url": "",
+    "data_type": "",
+    "doc_link": "",
+    "endpoint_link": "",
+    "num_followers": 0,
+    // "reviews": ""
   }
```
Comment (yohanyi) [inline source/public/javascripts/apiprofilepage.js]: WHAT THE RANDOM WHITESPACE!?

File: source/public/javascripts/apiprofilepage.js [hunk 7]
```diff
@@ -5,43 +5,44 @@ app.ApiProfile = {
...
 
-  // hit the database with the api_id sent in from the search page, do an ajax call to get the model shit we need
-  model: new app.ApiProfile.Models.Profile,
-
-  events: {
-    // event listeners for info tab, tips/tricks and links
-  },
-
-  initialize: function(){
-    console.log("Api Profile view initialized")
```
Comment (Lkroner) [inline source/public/javascripts/apiprofilepage.js:37]: take out console

**Pull request merged**